airflow/providers/redis/provider.yaml

```diff
@@ -48,7 +48,10 @@ versions:
 dependencies:
   - apache-airflow>=2.6.0
   # 5.0.2 excluded due to breaking changes which fixed in https://github.com/redis/redis-py/pull/3176
-  - redis>=4.5.2,!=4.5.5,!=5.0.2
+  # Limit redis to < 7.4 due to licencing change. In the future someone might contribute Valkey or any
+  # other fork support, and we might add support for it.
+  # https://redis.io/blog/redis-adopts-dual-source-available-licensing/
+  - redis>=4.5.2,!=4.5.5,!=5.0.2,<7.4
```

Comment:
I am torn on this one. Mainly because Redis will remain Redis from https://redis.io.

Once we have other forks, we should have a new provider for it, for example Valkey provider, similar to Elasticsearch & Hashicorp

Reply:
ah sorry right - we do not have to limit it. Redis client is actually 100% BSD and they have not changed it ... it's only the redis server that is impacted.